extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/JdbcMetaStoreManagerFactory.java

```diff
@@ -0,0 +1,296 @@
+      }
+    }
+
+    return Map.copyOf(results);
+  }
+
+  @Override
+  public Map<String, BaseResult> purgeRealms(Iterable<String> realms) {
+    Map<String, BaseResult> results = new HashMap<>();
+
+    for (String realm : realms) {
+      PolarisMetaStoreManager metaStoreManager = getOrCreateMetaStoreManager(() -> realm);
+      BasePersistence session = getOrCreateSessionSupplier(() -> realm).get();
+
+      PolarisCallContext callContext = new PolarisCallContext(session, diagServices);
+      BaseResult result = metaStoreManager.purge(callContext);
+      results.put(realm, result);
+
+      storageCredentialCacheMap.remove(realm);
+      sessionSupplierMap.remove(realm);
+      metaStoreManagerMap.remove(realm);
+    }
+
+    return Map.copyOf(results);
+  }
+
+  @Override
+  public synchronized PolarisMetaStoreManager getOrCreateMetaStoreManager(
+      RealmContext realmContext) {
+    if (!metaStoreManagerMap.containsKey(realmContext.getRealmIdentifier())) {
+      initializeForRealm(realmContext, null);
+      checkPolarisServiceBootstrappedForRealm(
+          realmContext, metaStoreManagerMap.get(realmContext.getRealmIdentifier()));
+    }
+    return metaStoreManagerMap.get(realmContext.getRealmIdentifier());
+  }
+
+  @Override
+  public synchronized Supplier<BasePersistence> getOrCreateSessionSupplier(
+      RealmContext realmContext) {
+    if (!sessionSupplierMap.containsKey(realmContext.getRealmIdentifier())) {
+      initializeForRealm(realmContext, null);
+      checkPolarisServiceBootstrappedForRealm(
+          realmContext, metaStoreManagerMap.get(realmContext.getRealmIdentifier()));
+    } else {
+      checkPolarisServiceBootstrappedForRealm(
+          realmContext, metaStoreManagerMap.get(realmContext.getRealmIdentifier()));
+    }
+    return sessionSupplierMap.get(realmContext.getRealmIdentifier());
+  }
+
+  @Override
+  public synchronized StorageCredentialCache getOrCreateStorageCredentialCache(
+      RealmContext realmContext) {
+    if (!storageCredentialCacheMap.containsKey(realmContext.getRealmIdentifier())) {
+      storageCredentialCacheMap.put(
+          realmContext.getRealmIdentifier(), new StorageCredentialCache());
+    }
+
+    return storageCredentialCacheMap.get(realmContext.getRealmIdentifier());
+  }
+
+  @Override
+  public synchronized EntityCache getOrCreateEntityCache(RealmContext realmContext) {
+    if (!entityCacheMap.containsKey(realmContext.getRealmIdentifier())) {
+      PolarisMetaStoreManager metaStoreManager = getOrCreateMetaStoreManager(realmContext);
+      entityCacheMap.put(realmContext.getRealmIdentifier(), new EntityCache(metaStoreManager));
+    }
+
+    return entityCacheMap.get(realmContext.getRealmIdentifier());
+  }
+
+  /**
+   * This method bootstraps service for a given realm: i.e. creates all the needed entities in the
+   * metastore and creates a root service principal. After that we rotate the root principal
+   * credentials and print them to stdout
```

Comment:
It does not look like this method prints anything to STDOUT :thinking:

Reply:
i didn't find either took it from LocamMetaStoreMangeFactory, removed the stdout part.